apps/prairielearn/python/prairielearn/core.py

```diff
@@ -1107,11 +1094,9 @@ def numpy_to_matlab_sf(A: _NumPyScalarType | npt.NDArray[Any], ndigits: int = 2)
     if np.isscalar(A):
         assert not isinstance(A, memoryview)
         if np.iscomplexobj(A):
-            if not isinstance(A, np.complexfloating) and isinstance(
-                A, np.generic | bytes | str
-            ):
-                raise TypeError(f"A must be a number, is {type(A)}")
-            scalar_str = _string_from_complex_sigfig(A, ndigits)
+            # `np.iscomplexobj` isn't a proper type guard, so we need to use
```

Comment:
My apologies, this was an oversight in what types `np.iscomplexobj` accepts. It does not seem intuitive that this would accept `np.float64`.

Reply:
Should probably still disallow `bytes` and `str` though?

Reply:
This isn't the issue; `isinstance(np.float64(...), np.generic) == True` so we were incorrectly reporting a "not a number" error for actual numbers.

Reply:
I added explicit checks for `str` and `bytes` in eaf4c1c.

Reply:
Ah, I should be referring to the branch on line 985 instead. This branch wasn't the one that immediately caused issues.